add App::help_subcommand_about() (close #1512)

yakubin opened the pull request:
I chose to name this method `help_subcommand_about()`, but I'm open to suggestions of better names.

rharriso commented:
Nice!

yakubin commented:
I've updated my commit to change the method name. Thank you for the kind words!

Dylan-DPC commented:
@yakubin you also need to rebase it with v3-master branch (which is the branch which will be soon be merged with master and become our 3.0 candidate)

yakubin commented:
> @yakubin you also need to rebase it with v3-master branch (which is the branch which will be soon be merged with master and become our 3.0 candidate)

It seems that this pull request isn't necessary after all. With `v3-master` the following works just fine:

```
let _ = App::new("customize")
        .version("0.1")
        .author("Nobody <odysseus@example.com>")
        .about("You can customize the version and help text")
        .arg(Arg::with_name("flag")
            .short('f')
            .long("flag")
            .help("Custom flag")
        )
        .arg(Arg::with_name("help")
            .short('H')
            .long("help")
            .help("Print help information")
        )
        .arg(Arg::with_name("version")
             .short('v')
             .long("version")
             .help("Print version information")
        )
        .subcommand(App::new("print").about("Print something"))
        .subcommand(App::new("help").about("Custom help subcommand"))
        .get_matches();
```

The subcommand still prints the help info as expected without any additional code. If this is correct, then this pull request may be closed without merging and the associated issue may be closed as invalid or fixed or whatever.

Notice that in the code above I have custom help and version flags without using `mut_arg()`. It still works as expected. It seems that the recommended way of doing it with `v3-master` is suboptimal?

Is it a desired behaviour that if I remove the flag `flag` from the code above, the usage lacks the `[FLAGS]` part?

spacekookie commented:
> @yakubin you also need to rebase it with v3-master branch (which is the branch which will be soon be merged with master and become our 3.0 candidate)

> It seems that this pull request isn't necessary after all. With `v3-master` the following works just fine:

Honestly, I wouldn't mind merging this to the current `master`, and
then doing a last release on the `2.0` branch for everybody who can't
or won't immediately upgrade to `3.0`. "Backporting" features like
this can be pretty nice, especially when the features are so smol.

Dylan-DPC commented:
@spacekookie i'm a bit divided on that because we haven't had a 2.x release for a while, and diverging to maintain 2 versions at this point given that we are close to rolling 3.0 over might not be worth the effort. 
(unrelated: check discord (#wg-cli) when you are free :P )

kbknapp commented:
With a change this small I'm ok merging onto v2-master and making a release. However, the caveat is v2 has no plans to be supported (short of maybe external contributions or security fixes).

@yakubin could you change the merge branch to v2-master?

yakubin commented:
> @yakubin could you change the merge branch to v2-master?

Done.

yakubin commented:
Just a note:

1. I rebased on top of `v2-master`.
2. I force-pushed the result to my GitHub fork.
3. I changed the merge branch in the GitHub UI.

AppVeyor was triggered between (2) and (3) and that's why the checks failed. Sorry if that complicates things. I haven't used this GitHub feature before, so I'm not sure how I should properly use it (i.e. in what order I should do the aforementioned steps).

Dylan-DPC commented:
Sorry for all the jumping, but I feel it's better to target this towards the 3.0 since we are getting close to it (that will also settle the CI issues) thanks

pksunkara commented:
If we are targetting to 3.0, then this PR is not needed as per https://github.com/clap-rs/clap/pull/1513#issuecomment-508608587.

If we are targetting to 2.0, then we have to replace appveyor with azure pipelines. I think we should do this anyway.

Regarding the PR itself, IMO I don't think it's needed for 2.0. If it's in maintenance mode, it should be purely bug fixes.

CreepySkeleton commented:
I agree with @pksunkara , only bugfixes  and docs for 2.x. There's no need to scatter our efforts for no good reason.
